internal/http/services/owncloud/ocdav/locks.go

```diff
@@ -190,11 +191,18 @@ func (cls *cs3LS) Create(ctx context.Context, now time.Time, details LockDetails
 	// see: http://www.webdav.org/specs/rfc2518.html#n-lock-tokens
 	token := uuid.New()
 
+	u := ctxpkg.ContextMustGetUser(ctx)
+
+	// add metadata via opaque
+	o := utils.AppendPlainToOpaque(nil, "lockownername", u.GetDisplayName())
+	o = utils.AppendPlainToOpaque(o, "locktime", now.Format(time.RFC3339))
+
 	r := &provider.SetLockRequest{
 		Ref: details.Root,
 		Lock: &provider.Lock{
-			Type: provider.LockType_LOCK_TYPE_EXCL,
-			User: details.UserID, // no way to set an app lock? TODO maybe via the ownerxml
+			Opaque: o,
+			Type:   provider.LockType_LOCK_TYPE_EXCL,
+			User:   details.UserID, // no way to set an app lock? TODO maybe via the ownerxml
```

Comment:
I think wo should write a creation time for the lock in the storage driver. A creation date might however not be supported by every storage provider, so this is optional and the web UI will need to be able to deal with missing info here . Same goes for the `ownername` property. It might be missing and the client then needs to fetch (and cache that user metadata).

Reply:
The client can't fetch data for a userID as a normal user is not allowed to get users by ID. Only admins can. 

Of course it is always wise for the webui to have a workaround if fields are missing. 

We create the expiration time also in the ocdav handler. So I put the creation time also here to have it consistent. As of now the storage provider just stores the lock data and doesn't add any additional information. Breaking this would mean there a two different sources of truth for the lock details: The ocdav handler AND the storage provider which is confusing.

Reply:
Yes, but in contrast to the expiration time the creation time should be determined automatically when writing the lock. The Lock() call can be made by other services than ocdav, eg. the wopiserver. 

WebDAV actually sets a timeout using a timespan, eg `Second-604800` (see https://greenbytes.de/tech/webdav/draft-reschke-webdav-locking-latest.html#rfc.section.2.5.7) and that needs to be relative to the creation date (and requires two propertios to be stored). So we decided back then that calculating the expiry time and storing only one datetime made more sense.

But we do loose the creation time. So, storing the creation date and the timeout in seconds seperately allow use to capture more details. @glpatcern I think we should reconsider and add this to the next iteration of the CS3 API.

Reply:
Well, after all the storage provider could already store all properties on `SetLock` (`creationdate = now()`, and `timeout = expirationdate - now()`). But I see you may want to expose both values when getting the Lock. Not sure how much it adds though...

Reply:
I added a ticket to the cs3api: https://github.com/cs3org/cs3apis/issues/213

We can do further discussion there.